File: backend/TravelPlanService/Models/Entities/TravelPlan.cs
namespace TravelPlanService.Models.Entities;

public class TravelPlan
{
    public long Id { get; set; }
    public long UserId { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public DateTime StartDate { get; set; }
    public DateTime EndDate { get; set; }
    public decimal Budget { get; set; }
    public string? Notes { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public ICollection<Destination> Destinations { get; set; } = new List<Destination>();
    public ICollection<Activity> Activities { get; set; } = new List<Activity>();
    public ICollection<Expense> Expenses { get; set; } = new List<Expense>();
    public ICollection<ChecklistItem> ChecklistItems { get; set; } = new List<ChecklistItem>();
    public ICollection<ShareToken> ShareTokens { get; set; } = new List<ShareToken>();
}

File: frontend/src/models/TravelPlan.ts
import type { Expense } from './Expense';

export interface TravelPlan {
  id: number;
  userId: number;
  name: string;
  description?: string;
  startDate: string;
  endDate: string;
  budget: number;
  totalExpenses?: number;
  remainingBudget?: number;
  notes?: string;
  createdAt: string;
  destinations?: Destination[];
  activities?: Activity[];
  expenses?: Expense[];
  checklistItems?: ChecklistItem[];
}

export interface AdminTravelPlan {
  id: number;
  userId: number;
  name: string;
  startDate: string;
  endDate: string;
  budget: number;
  totalExpenses: number;
  createdAt: string;
}

export interface Destination {
  id: number;
  travelPlanId: number;
  name: string;
  location: string;
  arrivalDate: string;
  departureDate: string;
  description?: string;
}

export interface DestinationRequestDto {
  name: string;
  location: string;
  arrivalDate: string;
  departureDate: string;
  description?: string;
}

export interface Activity {
  id: number;
  travelPlanId: number;
  destinationId?: number;
  name: string;
  date: string;
  time: string;
  location?: string;
  latitude?: number;
  longitude?: number;
  description?: string;
  estimatedCost?: number;
  status: 'planned' | 'reserved' | 'completed' | 'cancelled';
}

export interface ActivityRequestDto {
  destinationId?: number;
  name: string;
  date: string;
  time: string;
  location?: string;
  latitude?: number;
  longitude?: number;
  description?: string;
  estimatedCost?: number;
  status: Activity['status'];
}

export interface ChecklistItem {
  id: number;
  travelPlanId: number;
  name: string;
  isCompleted: boolean;
}

export interface ChecklistItemRequestDto {
  name: string;
  isCompleted: boolean;
}

export interface ShareToken {
  id: number;
  travelPlanId: number;
  token: string;
  accessType: 'VIEW' | 'EDIT';
  createdAt: string;
  expiresAt?: string;
}

export interface ShareTokenRequestDto {
  accessType: 'VIEW' | 'EDIT';
  expiresAt?: string;
}

export interface SharedPlanAccess {
  accessType: 'VIEW' | 'EDIT';
}
